internal/pkg/api/handleAck.go

```diff
@@ -215,20 +237,19 @@ func (ack *AckT) handleAckEvents(ctx context.Context, zlog zerolog.Logger, agent
 	}
 
 	for n, ev := range events {
+		event, _ := ev.AsGenericEvent()
 		span, ctx := apm.StartSpan(ctx, "ackEvent", "process")
 		span.Context.SetLabel("agent_id", agent.Agent.ID)
-		span.Context.SetLabel("action_id", ev.ActionId)
+		span.Context.SetLabel("action_id", event.ActionId)
 		log := zlog.With().
-			Str("actionType", string(ev.Type)).
-			Str("actionSubType", string(ev.Subtype)).
```

Comment:
I'm taking the opportunity to remove deprecated/unused attributes from our codebase (`stream_id` and `policy_id` have been fully removed)